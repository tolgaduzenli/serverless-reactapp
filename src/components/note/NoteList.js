import React from 'react'
import Note from './Note'

class NoteList extends React.Component {
    render() {
        const { noteList, onClickDelete, onClickEdit, onClickAdd } = this.props
        return <div>
            <table className="responsive-table centered striped">
                <thead>
                    <tr>
                        <th>Title</th>
                        <th>Description</th>
                        <th>Action</th>
                    </tr>
                </thead>
                <tbody>
                    {noteList && noteList.map((note, index) => {
                        return <Note key={index} note={note} onClickDelete={() => onClickDelete(note)} onClickEdit={() => onClickEdit(note)} />
                    })}
                </tbody>
            </table>
            <div className="fixed-action-btn">
                <button className="btn-floating btn-large waves-effect waves-light green" onClick={onClickAdd}>
                    <i className="material-icons">add</i></button>
            </div>
        </div>
    }
}

export default NoteList