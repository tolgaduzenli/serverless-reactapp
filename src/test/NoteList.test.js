import React from 'react'
import enzyme from 'enzyme'
import NoteList from '../components/note/NoteList'

describe('<NoteList />', () => {
    let wrapper
    let note = {
        title: "",
        description: "",
        _id: 1
    }
    
    beforeEach(() => {
        wrapper = enzyme.shallow(
            <NoteList noteList={[]}
                onClickAdd={jest.fn()}
                onClickDelete={jest.fn()}
                onClickEdit={jest.fn()}
            />
        )
    })

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('Should render a <table />', () => {
        expect(wrapper.find('table').length).toEqual(1)
    })

    it('Should render a <tbody />', () => {
        expect(wrapper.find('tbody').length).toEqual(1)
    })
    
    it('Should render a <button />', () => {
        expect(wrapper.find('button').length).toEqual(1)
    })

    it('should render an instance of the Note component for each Note object', () => {
        const noteList = []
        noteList.push(note)
        wrapper.setProps({ noteList });
        expect(wrapper.find('Note').length).toEqual(1);
    });
})