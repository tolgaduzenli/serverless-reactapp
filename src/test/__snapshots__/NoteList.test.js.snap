// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoteList /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteList
    noteList={Array []}
    onClickAdd={[MockFunction]}
    onClickDelete={[MockFunction]}
    onClickEdit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <table
          className="responsive-table centered striped"
        >
          <thead>
            <tr>
              <th>
                Title
              </th>
              <th>
                Description
              </th>
              <th>
                Action
              </th>
            </tr>
          </thead>
          <tbody />
        </table>,
        <div
          className="fixed-action-btn"
        >
          <button
            className="btn-floating btn-large waves-effect waves-light green"
            onClick={[MockFunction]}
          >
            <i
              className="material-icons"
            >
              add
            </i>
          </button>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  Title
                </th>
                <th>
                  Description
                </th>
                <th>
                  Action
                </th>
              </tr>
            </thead>,
            <tbody />,
          ],
          "className": "responsive-table centered striped",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Title
                </th>
                <th>
                  Description
                </th>
                <th>
                  Action
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Title
                  </th>,
                  <th>
                    Description
                  </th>,
                  <th>
                    Action
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Title",
                  },
                  "ref": null,
                  "rendered": "Title",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Description",
                  },
                  "ref": null,
                  "rendered": "Description",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Action",
                  },
                  "ref": null,
                  "rendered": "Action",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn-floating btn-large waves-effect waves-light green"
            onClick={[MockFunction]}
          >
            <i
              className="material-icons"
            >
              add
            </i>
          </button>,
          "className": "fixed-action-btn",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="material-icons"
            >
              add
            </i>,
            "className": "btn-floating btn-large waves-effect waves-light green",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "add",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "add",
            "type": "i",
          },
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <table
            className="responsive-table centered striped"
          >
            <thead>
              <tr>
                <th>
                  Title
                </th>
                <th>
                  Description
                </th>
                <th>
                  Action
                </th>
              </tr>
            </thead>
            <tbody />
          </table>,
          <div
            className="fixed-action-btn"
          >
            <button
              className="btn-floating btn-large waves-effect waves-light green"
              onClick={[MockFunction]}
            >
              <i
                className="material-icons"
              >
                add
              </i>
            </button>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                    Title
                  </th>
                  <th>
                    Description
                  </th>
                  <th>
                    Action
                  </th>
                </tr>
              </thead>,
              <tbody />,
            ],
            "className": "responsive-table centered striped",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    Title
                  </th>
                  <th>
                    Description
                  </th>
                  <th>
                    Action
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Title
                    </th>,
                    <th>
                      Description
                    </th>,
                    <th>
                      Action
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Title",
                    },
                    "ref": null,
                    "rendered": "Title",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Description",
                    },
                    "ref": null,
                    "rendered": "Description",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Action",
                    },
                    "ref": null,
                    "rendered": "Action",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn-floating btn-large waves-effect waves-light green"
              onClick={[MockFunction]}
            >
              <i
                className="material-icons"
              >
                add
              </i>
            </button>,
            "className": "fixed-action-btn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="material-icons"
              >
                add
              </i>,
              "className": "btn-floating btn-large waves-effect waves-light green",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "add",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "add",
              "type": "i",
            },
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
